File: tests/Fixtures/csharp/Shop/Stock/Picking.cs
namespace Shop.Stock;

// A pick list skips an empty bin with `continue` — then puts the real work in an `else` anyway, one
// level deeper than the loop needs.
public static class Picking
{
    public static IReadOnlyList<string> List(IReadOnlyDictionary<string, int> bins)
    {
        var picks = new List<string>();

        foreach (var (bin, units) in bins)
        {
            // @sin RedundantElse
            if (units == 0)
            {
                continue;
            }
            else
            {
                picks.Add($"{bin}: {units}");
                picks.Sort(StringComparer.Ordinal);
            }
        }

        return picks;
    }
}

File: tests/Fixtures/python/shop/picking.py
# Building the pick list skips the lines already picked — then puts the real work in an `else:`
# after the `continue`.


def pick_list(orders) -> list:
    picks = []
    for order in orders:
        for line in order.lines:
            # @sin RedundantElse
            if line.picked:
                continue
            else:
                picks.append((order.number, line.sku, line.quantity))
    return picks
